aten/src/ATen/native/vulkan/ops/BinaryElementwise.cpp

```diff
@@ -0,0 +1,432 @@
+#include <ATen/native/vulkan/ops/Common.h>
+#include <torch/library.h>
+
+namespace at {
+namespace native {
+namespace vulkan {
+namespace ops {
+namespace {
+
+using namespace api::utils;
+
+void check_inputs(const Tensor& input1, const Tensor& input2) {
+  TORCH_CHECK(
+      channels_size(input1) == channels_size(input2),
+      "Vulkan binary elementwise ops require channel dimension to be equal!");
+  if (batch_size(input1) != batch_size(input2)) {
+    TORCH_CHECK(
+        channels_size(input1) % 4 == 0,
+        "Vulkan binary elementwise ops require channel to be a multiple of 4 to broadcast along batch dimension!")
+  }
```

Comment:
Please consider using more vertical spacing in your functions.  It makes code easier to read.  I generally group related stuff together but try not to put more than four or five lines next to one another.  Makes reading the code a lot easier.